### PR TITLE
Check modules and update on re-render

### LINKED ISSUE #42: DynamicModuleLoader will not respond to module updates
If the component is re-rendered with different modules, they will not be added as we do not listen to componentWillReceiveProps

### PR DESCRIPTION
**Pull request by abettadapur**
Fixes #42

### CODE REVIEW

File: packages/redux-dynamic-modules/src/DynamicModuleLoader.tsx [hunk 1]
```diff
@@ -102,7 +102,30 @@ class DynamicModuleLoaderImpl extends React.Component<
             }
         }
 
-        this._addedModules = store.addModules(modules);
+        for(const module of modules) {
+            this._addedModules[module.id] = store.addModule(module);
+        }
+    }
+
+    public getSnapshotBeforeUpdate() {
```
Comment (cschleiden) [inline packages/redux-dynamic-modules/src/DynamicModuleLoader.tsx]: Isn't this too late? Like do we want to make sure if this component is re-rendered with additional modules that they are immediately available to children in the tree?

Comment (abettadapur) [reply packages/redux-dynamic-modules/src/DynamicModuleLoader.tsx]: @cschleiden@live.de Do you know of a lifecyle method where I can do this now?

---
In reply to: [246155351](https://github.com/Microsoft/redux-dynamic-modules/pull/43#discussion_r246155351) [](ancestors = 246155351)

Comment (cschleiden) [reply packages/redux-dynamic-modules/src/DynamicModuleLoader.tsx]: @abettadapur I couldn't find a good solution. A hacky one might be to put the store in the state and then use `getDerivedStateFromProps`, or do it in `render`, memoized in some way

**Comment by navneet-g** [comment]
Do you plan to finish this?

**Comment by abettadapur** [comment]
@navneet-g I don't have any ideas on how to address the feedback Chris mentioned above. Now that they have taken our pre-render hook. 
On another note, we have a similar problem with ReactRedux6 wherein module data may not exist in the component during the first render. Perhaps we should come up with a mechanism to solve both of these issues.

Perhaps we can:
* Perform some checks in DML to not render children if all module dependencies have not been initialized. I guess this would involve checking the state shape to ensure it contains all of the keys defined by the `reducerMap`s.
* Use React.Suspense to accomplish the same task for newer versions of React.

What do you think?

**Pull request closed without merging**